zboxcore/sdk/chunked_upload.go

```diff
@@ -594,12 +597,24 @@ func (su *ChunkedUpload) processUpload(chunkStartIndex, chunkEndIndex int,
 			defer wg.Done()
 			err = b.sendUploadRequest(ctx, su, chunkEndIndex, isFinal, encryptedKey, body, formData, pos)
 			if err != nil {
+				wgErrors <- err
 				logger.Logger.Error("error during sendUploadRequest", err)
 			}
 		}(blobber, body, formData, pos)
 	}
 
-	wg.Wait()
+	go func() {
+		wg.Wait()
+		close(wgDone)
+	}()
+	
+	select {
+		case <-wgDone:
+			break
+		case err := <-wgErrors:
+			close(wgErrors)
```

Comment:
`wgErrors <- err` in line 600 above will panic in another goroutine after we close the `wgErrors` here. We should only close on data writer rather than the reader. Anyway, to fix it and avoid goroutine leak, we can use `select` in line 600, and remove the `close(wgErrors)` here.
```go
select {
wgErrors <- err:
default:
}
```

Reply:
Thanks for the input @peterlimg. Understood why the panic will happen and corrected the code accordingly.